### PR TITLE
BACK-1582 log context 2

### PR DESCRIPTION
**Pull request by estrauser-ledger**
## What is this about?

Still more logs.
Kinda changed things a little bit to allow for context in the Account Manager

## Cute picture of lama
0911llama1
_Un LamaLog dans son LoggingContext de terrain de golf_

### CODE REVIEW

File: coins/bitcoin/api/src/main/scala/co/ledger/lama/bitcoin/api/middlewares/LoggingMiddleware.scala [hunk 1]
```diff
@@ -13,8 +13,8 @@ object LoggingMiddleware extends DefaultContextLogging {
           resp
         case resp =>
           log.error(s"""
-               |A ${resp.status} error occurred: ${resp.body}
-               |Request was : $req""".stripMargin)
+               A ${resp.status} error occurred: ${resp.body}
+               Request was : $req""")
```
Comment (gagbo) [inline coins/bitcoin/api/src/main/scala/co/ledger/lama/bitcoin/api/middlewares/LoggingMiddleware.scala:17]: You don’t need stripMargin anymore ?

Comment (estrauser-ledger) [reply coins/bitcoin/api/src/main/scala/co/ledger/lama/bitcoin/api/middlewares/LoggingMiddleware.scala:17]: I hate stripMargin..

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala [hunk 2]
```diff
@@ -35,26 +38,29 @@ object CursorStateService {
        * - An unknown valid hash return either a 404, or an empty list
        * - An invalid hash returns either a 400 or a 500 error.
        */
+
       explorerClient
         .getBlock(block.hash)
         .flatMap {
           case Some(lvb) => IO.pure(lvb)
-          case None      => fetchLastBlocksUntilValid(accountId, block)
+          case None      => fetchLastBlocksUntilValid(account.id, block)
```
Comment (gagbo) [inline coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala:46]: That'd be cool to log that a probable reorg has been detected and a rollback is occurring

File: common/src/main/scala/co/ledger/lama/common/logging/LamaLogContext.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import co.ledger.lama.common.models.{Account, AccountGroup, Coin, CoinFamily}
+
+case class LamaLogContext(
+    accountId: Option[UUID] = None,
+    identifier: Option[String] = None,
+    coinFamily: Option[CoinFamily] = None,
+    coin: Option[Coin] = None,
+    group: Option[AccountGroup] = None,
+    followUpId: Option[UUID] = None,
+    customFields: List[Option[(String, String)]] = List()
```
Comment (gagbo) [inline common/src/main/scala/co/ledger/lama/common/logging/LamaLogContext.scala:14]: Using a map here would be better to ensure that a key is present only once

**Review by gagbo — APPROVED** [review]
Seems good, I think that using a map for the custom fields of LamaContext would be better, but that's definitely not a blocker :tada:

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala (hunk 2 above)
Comment (estrauser-ledger) [reply coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala:46]: Ye we'll have to check everywhere we can add useful log

**Pull request merged**